test/ts/unit/contract_registry.ts

```diff
@@ -0,0 +1,195 @@
+
+        const contractRegistryTruffle = await contractRegistryArtifact.new(
+            collateralizer.address,
+            debtKernel.address,
+            mockDebtRegistry.address,
+            mockDebtToken.address,
+            repaymentRouter.address,
+            mockTokenRegistry.address,
+            mockTokenTransferProxy.address,
+            { from: CONTRACT_OWNER },
+        );
+
+        const contractRegistryAsWeb3Contract = web3.eth
+            .contract(contractRegistryArtifact.abi)
+            .at(contractRegistryTruffle.address);
+
+        contractRegistry = new ContractRegistryContract(
+            contractRegistryAsWeb3Contract,
+            TX_DEFAULTS,
+        );
+
+        ABIDecoder.addABI(contractRegistry.abi);
+    });
+
+    after(() => {
+        ABIDecoder.removeABI(contractRegistry.abi);
+    });
+
+    describe("Initialization", () => {
+        it("points to the collateralizer passed in through the constructor", async () => {
+            await expect(contractRegistry.collateralizer.callAsync()).to.eventually.equal(
+                collateralizer.address,
+            );
+        });
+        it("points to the debt kernel passed in through the constructor", async () => {
+            await expect(contractRegistry.debtKernel.callAsync()).to.eventually.equal(
+                debtKernel.address,
+            );
+        });
+        it("points to the debt registry passed in through the constructor", async () => {
+            await expect(contractRegistry.debtRegistry.callAsync()).to.eventually.equal(
+                mockDebtRegistry.address,
+            );
+        });
+        it("points to the debt token passed in through the constructor", async () => {
+            await expect(contractRegistry.debtToken.callAsync()).to.eventually.equal(
+                mockDebtToken.address,
+            );
+        });
+        it("points to the repayment router passed in through the constructor", async () => {
+            await expect(contractRegistry.repaymentRouter.callAsync()).to.eventually.equal(
+                repaymentRouter.address,
+            );
+        });
+        it("points to the token registry passed in through the constructor", async () => {
+            await expect(contractRegistry.tokenRegistry.callAsync()).to.eventually.equal(
+                mockTokenRegistry.address,
+            );
+        });
+        it("points to the token transfer proxy passed in through the constructor", async () => {
+            await expect(contractRegistry.tokenTransferProxy.callAsync()).to.eventually.equal(
+                mockTokenTransferProxy.address,
+            );
+        });
+    });
+
+    describe("#updateAddress", () => {
+        describe("successfully", () => {
+            let txHash: string;
+
+            before(async () => {
+                txHash = await contractRegistry.updateAddress.sendTransactionAsync(
+                    new BigNumber(2),
```

Comment:
Nit: please create a constant called `DEBT_REGISTRY_ENUM_ID` or something to that effect instead of `new BigNumber(2)`